avsomr/avsomrlocal.cpp

```diff
@@ -0,0 +1,289 @@
+#include "avsomrlocal.h"
+
+#include <QFileInfo>
+#include <QEventLoop>
+#include <QProcess>
+
+#include "mscore/preferences.h"
+
+#include "avslog.h"
+#include "avsomrlocalinstaller.h"
+
+namespace {
+
+static const QString AVS_EXT_NAME("avsomr_local");
+
+#ifdef Q_OS_WIN
+
+static const QString SHELL("cmd");
+static const QStringList SHELL_ARGS("/c");
+static const QString AVS_MAKE_SCRIPT("make_avs_win.bat");
+
+#elif defined(Q_OS_MAC)
+
+static const QString SHELL("sh");
+static const QStringList SHELL_ARGS;
+static const QString AVS_MAKE_SCRIPT("make_avs_mac.sh");
+
+#else
+
+static const QString SHELL("sh");
+static const QStringList SHELL_ARGS;
+static const QString AVS_MAKE_SCRIPT("make_avs_lin.sh");
+
+#endif
+}
+
+using namespace Ms::Avs;
+
+AvsOmrLocal::AvsOmrLocal()
```

Comment:
Better in the .h?

Reply:
If this is not important, then I prefer to leave the constructors and destructors of classes in the .cpp file, excluding interfaces, data types without logic (and without .cpp file)  and small child structures.
I usually do this so as not to clutter up the .h files (it should be well readable), and for debugging purposes, it’s easier to add a debugger breakpoint, just add logging to create and delete an object.

Reply:
hence my ?

Reply:
This class not interface, has logic (and .cpp file), not a small child of another class, therefore, its constructor and destructor in the .cpp